test/e2e/elasticache/replication_group/coverage_tests/rg_creation.sh

```diff
@@ -0,0 +1,245 @@
+  automatic_failover_enabled="true"
+  num_node_groups=1
+  replicas_per_node_group=1
+  multi_az_enabled="false"
+  yaml_base="$(provide_replication_group_yaml)"
+  rg_yaml=$(cat <<EOF
+$yaml_base
+    nodeGroupConfiguration:
+      - nodeGroupID: 1020
+        primaryAvailabilityZone: us-east-1a
+EOF
+)
+  output_msg=$(echo "$rg_yaml" | kubectl apply -f - 2>&1)
+  error_code=$?
+
+  # kubectl apply should fail if node group ID is not placed within quotes
+  if [ $error_code -eq 0 ]; then
+    echo "FAIL: expected config application for replication group $rg_id to have failed"
+    exit 1
+  fi
+
+  # check that error message is the one we expect
+  if [[ $output_msg != *"spec.nodeGroupConfiguration.nodeGroupID: Invalid value"*  ]]; then
+    echo "FAIL: creation of replication group $rg_id failed as expected, but error message different than expected:"
+    echo "$output_msg"
+    exit 1
+  fi
+}
+
+# create replication group with custom nodeGroupConfiguration
+test_create_rg_custom_node_config() {
+  debug_msg "executing ${FUNCNAME[0]}"
+
+  # generate and apply yaml for replication group creation
+  clear_rg_parameter_variables
+  rg_id="rg-custom-node-config"
+  num_node_groups=1
+  replicas_per_node_group=2
+  yaml_base="$(provide_replication_group_yaml)"
+  rg_yaml=$(cat <<EOF
+$yaml_base
+    nodeGroupConfiguration:
+      - nodeGroupID: "0010"
+        primaryAvailabilityZone: us-east-1a
+        replicaAvailabilityZones:
+          - us-east-1b
+          - us-east-1a
+EOF
+)
+  output_msg=$(echo "$rg_yaml" | kubectl apply -f - 2>&1)
+  exit_if_rg_config_application_failed $? "$rg_id"
+
+  # ensure resource successfully created and available
+  wait_and_assert_replication_group_available_status
+}
+
+# create RG with custom param group
+test_create_rg_custom_param_group() {
+  debug_msg "executing ${FUNCNAME[0]}"
+
+  # create custom param group
+  k8s_controller_reload_credentials "elasticache"
+  daws elasticache create-cache-parameter-group --cache-parameter-group-name "pgtest" --cache-parameter-group-family "redis6.x" --description "test" 1>/dev/null 2>&1
+  daws elasticache modify-cache-parameter-group --cache-parameter-group-name "pgtest" --parameter-name-values "ParameterName=reserved-memory-percent,ParameterValue=30" 1>/dev/null 2>&1
+
+  # generate and apply yaml for replication group creation
+  clear_rg_parameter_variables
+  rg_id="rg-custom-param-group"
+  num_node_groups=1
+  replicas_per_node_group=2
+  yaml_base="$(provide_replication_group_yaml)"
+  rg_yaml=$(cat <<EOF
+$yaml_base
+    cacheParameterGroupName: pgtest
+EOF
+)
+  output_msg=$(echo "$rg_yaml" | kubectl apply -f - 2>&1)
+  exit_if_rg_config_application_failed $? "$rg_id"
+
+  # ensure resource successfully created and available, check resource is as expected
+  wait_and_assert_replication_group_available_status
+  aws_assert_rg_param_group "$rg_id" "pgtest"
+}
+
+# create multiple RGs and check deletion succeeds
+test_rg_deletion_multiple() {
```

Comment:
Is this really a create failure test? If not, how about putting this into a separate file, maybe call it `test/e2e/elasticache/replication_group/create_multi.sh`?

Reply:
It's not really a creation test nor is it a negative test, it was just a test for deleting multiple resources at once. It's our only test scenario right now that deals with deletion so I felt a little awkward putting it in its own file. Maybe I'll place it in the miscellaneous tests